src/js/component.js

```diff
@@ -432,7 +431,8 @@ class Component {
     componentName = componentName || (component.name && toTitleCase(component.name()));
 
     if (componentName) {
-      this.childNameIndex_[componentName] = component;
+      this.childNameIndex_[toTitleCase(componentName)] = component;
```

Comment:
Is this needed? The behavior related to this isn't changing. Same with line 484.

Reply:
We need this because, `getChild` used to `toTitleCase` the `name` that it was given. Now we won't do that because we register both possible names during `addChild`.

Reply:
oh, didn't realize that  getChild used this property. However, it sounds like we don't need to call `toTitleCase` again because it sounds like `componentName` is always `toTitleCase`d per lines 430 and 388. So, we would only need to register the toLowerCase version.

Reply:
👍